### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by badb**
https://github.com/imapp-pl/golem/wiki/Logging

Resolves: #428

### CODE REVIEW

File: gnr/main.py [hunk 1]
```diff
@@ -12,7 +13,10 @@
 @click.command()
 @click.option('--datadir', '-d', type=click.Path())
 def main(datadir):
-    config_logging()
+    if datadir:
```
Comment (chfast) [inline gnr/main.py]: Can you move it to golemapp.start()?

File: gnr/logging.ini [hunk 2]
```diff
@@ -1,28 +1,39 @@
...
+level: WARNING
+handlers: consoleHandler
+
+[logger_golem]
+level: INFO
+qualname: golem
+handlers: fileHandlerWithTimeRot
+
+[logger_pay]
+level: DEBUG
```
Comment (chfast) [inline gnr/logging.ini]: golem.pay does not have to be DEBUG. Is it here as an example?

File: tests/gnr/task/test_gnrtask.py [hunk 3]
```diff
@@ -29,6 +29,7 @@ def test_gnr_task(self):
             self.assertEqual(task.get_results(subtask_id), False)
         self.assertTrue(any(["not my subtask" in log for log in l.output]))
 
+
```
Comment (chfast) [inline tests/gnr/task/test_gnrtask.py:22]: Ooops.

**Comment by chfast** [comment]
Level of golem.pay is to high by default. Can be merged after fixed.

File: gnr/logging.ini (hunk 2 above)
Comment (badb) [inline gnr/logging.ini]: Should I comment it out or remove?

Comment (chfast) [inline gnr/logging.ini]: Change to INFO.

**Comment by chfast** [comment]
After merging in `handle_key_error` changes I have problem with `test_gnrtask.py`. I'm not sure which version is final as the same changes are applied many times in different branches. Anyway, functions like `GNRTask.get_stdout()` do not throw KeyErrors but have `handle_key_error` modifier.

Please remove my last two commits if needed.

**Comment by badb** [comment]
Fixed. Those functions should not have `handle_key_error` modifier (subtask_id is checked on the higher level).

**Pull request merged**